### PR TITLE
Pass view owner to getViewRunAsIdentity 

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

**Comment by kokosing** [comment]
Passing just user is not enough as groups are needed too.

**Comment by kokosing** [comment]
> Passing just user is not enough as groups are needed too.

Groups are not problem as one can use `GroupProvider`

**Review by ksobolew — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java [hunk 1]
```diff
@@ -1190,11 +1195,18 @@ public boolean isView(Session session, QualifiedObjectName viewName)
 
         Optional<String> owner = connectorView.get().getOwner();
         Identity runAsIdentity = systemSecurityMetadata.getViewRunAsIdentity(session, viewName.asCatalogSchemaTableName(), owner)
-                .or(() -> owner.map(Identity::ofUser))
+                .or(() -> owner.map(this::createViewOwnerIdentity))
                 .orElseThrow(() -> new TrinoException(NOT_SUPPORTED, "Catalog does not support run-as DEFINER views: " + viewName));
```
Comment (ksobolew) [inline core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java:1199]: Just a note that this error message doesn't make sense to me. The view is `SECURITY DEFINER` and there's no owner - looks more like metadata inconsistency?

Comment (kokosing) [reply core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java:1199]: 👍

**Review by Praveen2112 — APPROVED** [review]

**Pull request closed without merging**